parquet/src/main/java/org/apache/iceberg/parquet/ParquetFilters.java

```diff
@@ -38,9 +43,71 @@
 
 class ParquetFilters {
 
+  private static final Set<Operation> SUPPORTED_OPS = ImmutableSet.of(
+          Operation.IS_NULL,
+          Operation.NOT_NULL,
+          Operation.EQ,
+          Operation.NOT_EQ,
+          Operation.GT,
+          Operation.GT_EQ,
+          Operation.LT,
+          Operation.LT_EQ);
+
+  private static final Set<Type.TypeID> SUPPORTED_TYPES = ImmutableSet.of(
+          Type.TypeID.BOOLEAN,
+          Type.TypeID.INTEGER,
+          Type.TypeID.LONG,
+          Type.TypeID.FLOAT,
+          Type.TypeID.DOUBLE,
+          Type.TypeID.DATE,
+          Type.TypeID.TIME
+  );
+
   private ParquetFilters() {
   }
 
+  public static boolean isSupportedFilter(Expression expr, Schema schema, boolean caseSensitive) {
```

Comment:
That is what I originally use but I changed it because throwing/catching exception is pretty expensive. In the non-error case, we should avoid that, although I understand creating some separate code to keep tracking is another pain.

Reply:
Gotcha! Since this is not in the critical loop of reading rows, I think the cost of catching exception should not be significant in the grand scheme of things. I don't have a strong opinion for this though.

Reply:
We throw it once per file in case of unsupported filters. I don't have strong opinions either.